django/db/models/fields/related.py

```diff
@@ -528,6 +528,11 @@ def _check_unique_target(self):
             frozenset(ut)
             for ut in self.remote_field.model._meta.unique_together
         })
+        if self.remote_field.model._meta.constraints:
+            for constraint in self.remote_field.model._meta.constraints:
```

Comment:
You should omit partial unique constraints, please use [`total_unique_constraints()`](https://github.com/django/django/blob/0352a44dd61c19bebf0c0b305dbbc3f710ff9945/django/db/models/options.py#L830-L840).